src/PathProcessor/TideSitePathProcessor.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace Drupal\tide_site\PathProcessor;
+
+use Drupal\node\Entity\Node;
+use Drupal\path_alias\AliasManagerInterface;
+use Drupal\Core\PathProcessor\InboundPathProcessorInterface;
+use Drupal\Core\PathProcessor\OutboundPathProcessorInterface;
+use Drupal\Core\Render\BubbleableMetadata;
+use Drupal\tide_site\AliasStorageHelper;
+use Drupal\tide_site\TideSiteHelper;
+use Symfony\Component\DependencyInjection\ContainerAwareTrait;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Path processor for tide_site module.
+ */
+class TideSitePathProcessor implements InboundPathProcessorInterface, OutboundPathProcessorInterface {
+  use ContainerAwareTrait;
+
+  /**
+   * An alias manager for looking up the system path.
+   *
+   * @var \Drupal\path_alias\AliasManagerInterface
+   */
+  protected $aliasManager;
+
+  /**
+   * Tide site helper.
+   *
+   * @var \Drupal\tide_site\TideSiteHelper
+   */
+  protected $tideSiteHelper;
+
+  /**
+   * Tide path alias helper.
+   *
+   * @var \Drupal\tide_site\AliasStorageHelper
+   */
+  protected $tidAliasHelper;
+
+  /**
+   * Constructs a PathProcessorAlias object.
+   *
+   * @param \Drupal\path_alias\AliasManagerInterface $alias_manager
+   *   An alias manager for looking up the system path.
+   * @param \Drupal\tide_site\TideSiteHelper $siteHelper
+   *   Tide site helper.
+   * @param \Drupal\tide_site\AliasStorageHelper $aliasStorageHelper
+   *   Tide path alias helper.
+   */
+  public function __construct(AliasManagerInterface $alias_manager, TideSiteHelper $siteHelper, AliasStorageHelper $aliasStorageHelper) {
+    $this->aliasManager = $alias_manager;
+    $this->tideSiteHelper = $siteHelper;
+    $this->tideAliasHelper = $aliasStorageHelper;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processInbound($path, Request $request) {
```

Comment:
@vincent-gao is this required by one of the interfaces? Is it doing anything different than the default implementation?

Reply:
@anthony-malkoun , No , it doesn't required. deleted. Thanks for your CR.